ttnn/cpp/ttnn/events.hpp

```diff
@@ -0,0 +1,40 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <memory>
+
+#include "tt_metal/impl/device/device_mesh.hpp"
+
+namespace ttnn {
+
+namespace events {
+
+struct MultiDeviceEvent
+{
+    MultiDeviceEvent(DeviceMesh* device_mesh) {
```

Comment:
Please move it to the c++.
Als it is a good thing to always put TT_ASSERT(device_mesh != nullptr);

Reply:
done